packages/nest/package.py

```diff
@@ -0,0 +1,179 @@
+            os.environ["USE_PROFILER_WRAPPER"] = "1"
+
+    def profiling_wrapper_off(self):
+        if self.spec.satisfies('+profile'):
+            del os.environ["USE_PROFILER_WRAPPER"]
+
+    def get_optimization_level(self):
+        flags = "-g"
+
+        if 'bgq' in self.spec.architecture:
+            flags += ' -O3'
+        else:
+            flags += ' -O2'
+
+        if self.spec.satisfies('+knl'):
+            flags = '-g -xmic-avx512 -O3 -qopt-report=5'
+
+        return flags
+
+    def install(self, spec, prefix):
+        build_dir = "spack-build-%s" % spec.version
+
+        with working_dir(build_dir, create=True):
+            c_compiler = spack_cc
+            cxx_compiler = spack_cxx
+
+            optflag = self.get_optimization_level()
+
+            if self.spec.satisfies('+profile'):
+                c_compiler = 'tau_cc'
+                cxx_compiler = 'tau_cxx'
+            # for bg-q, our cmake is not setup properly
+            elif 'bgq' in self.spec.architecture and self.spec.satisfies('+mpi'):
+                    c_compiler = self.spec['mpi'].mpicc
+                    cxx_compiler = self.spec['mpi'].mpicxx
+
+            cmake_options = ['-DCMAKE_INSTALL_PREFIX:PATH=%s' % prefix,
+                             '-DCMAKE_C_FLAGS=%s' % optflag,
+                             '-DCMAKE_CXX_FLAGS=%s' % optflag,
+                             '-DCMAKE_C_COMPILER=%s' % c_compiler,
+                             '-DCMAKE_CXX_COMPILER=%s' % cxx_compiler,
+                             '-Dwith-ltdl=OFF',
+                             '-Dwith-readline=OFF']
+
+            # not sure the -Dwith-optimize=flags option is required if we already set CMAKE_C_FLAGS above
+            # but just to be safe
+            semicolon_separated_optflag = optflag.translate(string.maketrans(' ', ';'))
+            cmake_options.extend(['-Dwith-optimize=' + semicolon_separated_optflag])
+
+            if self.spec.satisfies('~python'):
+                cmake_options.extend(['-Dwith-python=OFF'])
+                cmake_options.extend(['-Dcythonize-pynest=OFF'])
+            else:
+                cmake_options.extend(['-Dwith-python=ON'])
+                cmake_options.extend(['-Dcythonize-pynest=%s' % self.spec['py-cython'].prefix])
+
+            if self.spec.satisfies('~gsl'):
+                cmake_options.extend(['-Dwith-gsl=OFF'])
+            else:
+                cmake_options.extend(['-Dwith-gsl=%s' % self.spec['gsl'].prefix])
+
+            if self.spec.satisfies('~mpi'):
+                cmake_options.extend(['-Dwith-mpi=OFF'])
+            else:
+                cmake_options.extend(['-Dwith-mpi=ON'])
+
+            if self.spec.satisfies('~openmp'):
+                cmake_options.extend(['-Dwith-openmp=OFF'])
+            else:
+                cmake_options.extend(['-Dwith-openmp=ON'])
+
+            if self.spec.satisfies('~static'):
+                cmake_options.extend(['-Dstatic-libraries=OFF'])
```

Comment:
yesterday one of `dev` told me `shared` is more `preferred` name for variant instead of `static`